packages/mobile/e2e/src/usecases/WalletConnectV1.js

```diff
@@ -0,0 +1,324 @@
+    const message = hashMessageWithPrefix(`My email is valora.test@mailinator.com - ${+new Date()}`)
+    const msgParams = [fromAddress, message]
+    let result = walletConnector.signMessage(msgParams)
+    await waitFor(element(by.text('Sign a data payload')))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+    await element(by.text('Allow')).tap()
+    await enterPinUiIfNecessary()
+    await waitFor(element(by.text('Success! Please go back to WalletConnectV1 E2E to continue')))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+    await waitFor(element(by.id('SendOrRequestBar')))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+
+    // Wait for signature
+    let signature = await result
+
+    // Verify the signature
+    const valid = verifySignature(message, signature, fromAddress)
+    const invalid = verifySignature(message, signature, toAddress)
+    jestExpect(valid).toStrictEqual(true)
+    jestExpect(invalid).toStrictEqual(false)
+  })
+
+  // TODO: Investigate failing
+  it.skip('Then is able to sign typed data (eth_signTypedData)', async () => {
+    const typedData = {
+      types: {
+        EIP712Domain: [
+          { name: 'name', type: 'string' },
+          { name: 'version', type: 'string' },
+          { name: 'chainId', type: 'uint256' },
+          { name: 'verifyingContract', type: 'address' },
+        ],
+        Person: [
+          { name: 'name', type: 'string' },
+          { name: 'account', type: 'address' },
+        ],
+        Mail: [
+          { name: 'from', type: 'Person' },
+          { name: 'to', type: 'Person' },
+          { name: 'contents', type: 'string' },
+        ],
+      },
+      primaryType: 'Mail',
+      domain: {
+        name: 'Example Dapp',
+        version: '1.0',
+        chainId: 1,
+        verifyingContract: '0x0000000000000000000000000000000000000000',
+      },
+      message: {
+        from: {
+          name: 'Alice',
+          account: '0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+        },
+        to: {
+          name: 'Bob',
+          account: '0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb',
+        },
+        contents: 'Hey, Bob!',
+      },
+    }
+
+    const msgParams = [fromAddress, JSON.stringify(typedData)]
+
+    let result = walletConnector.signTypedData(msgParams)
+    await waitFor(element(by.text('Sign a data payload')))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+    await element(by.text('Allow')).tap()
+    await enterPinUiIfNecessary()
+    await waitFor(element(by.id('SendOrRequestBar')))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+
+    // Wait for signature
+    let signature = await result
+
+    // Verify the signature - currently both return valid
+    const valid = verifySignature(JSON.stringify(typedData), signature, fromAddress)
+    const invalid = verifySignature(JSON.stringify(typedData), signature, toAddress)
+    jestExpect(valid).toStrictEqual(true)
+    jestExpect(invalid).toStrictEqual(false)
```

Comment:
wondering if there's any difference between `jestExpect` and `expect` - i see `expect` in most other specs?

Reply:
Detox overwrites jest's expect with it's own. This reimports jest's expect under a different name.